src/survival.py

```diff
@@ -27,6 +29,17 @@ def n_best_survive(
     return list(_partition(population_with_fitness, [n])[:n])
 
 
+def n_best_and_m_random_survive(
+    population_with_fitness: list[tuple[Genotype, float]], n: int, m: int
+) -> list[tuple[Genotype, float]]:
+    if n + m > len(population_with_fitness):
+        raise ValueError("n + m > len(population_with_fitness)")
```

Comment:
N.O. Actually it is a valid case. You can't just raise Error. Write proper error handling

Reply:
I'm not sure why it is valid. I think that calling `n_best_survive` with a number greater than `len(population_with_fitness)` would also result in error:
`ValueError: kth(=-200) out of bounds (200)`

Reply:
`n` and `m` are hiperparameters. I see no reason to forbid population size to change during simulation (although we can)

Reply:
But it is not the purpose of this mutation. It was supposed to be a simple one.

Reply:
Just as `n_best_survive` :)

Reply:
it's not even mutation

Reply:
I meant survival function...

Reply:
It will explode. Mutators will create forbidden genotypes, sanitizers will remove them and this method will get too small population and raise Error